crates/matrix-sdk-crypto/src/machine/tests/olm_decryption_encryption_info.rs

```diff
@@ -0,0 +1,149 @@
+// Copyright 2020 The Matrix.org Foundation C.I.C.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use assert_matches2::{assert_let, assert_matches};
+use matrix_sdk_common::deserialized_responses::{
+    ProcessedToDeviceEvent, VerificationLevel, VerificationState,
+};
+use matrix_sdk_test::async_test;
+use ruma::TransactionId;
+use serde_json::json;
+
+use crate::{
+    machine::{
+        test_helpers::{
+            bootstrap_requests_to_keys_query_response, build_session_for_pair,
+            encrypt_to_device_helper, get_machine_pair_with_session,
+            get_prepared_machine_test_helper,
+        },
+        tests,
+        tests::{
+            alice_device_id, alice_id,
+            decryption_verification_state::mark_alice_identity_as_verified_test_helper,
+        },
+    },
+    verification::tests::bob_id,
+    DeviceData, OlmMachine,
+};
+
+#[async_test]
+async fn test_to_device_verification_state_signed_device() {
+    let (alice, bob) = get_machine_pair_with_session(alice_id(), tests::user_id(), false).await;
+
+    let bootstrap_requests = alice.bootstrap_cross_signing(false).await.unwrap();
+    let kq_response = bootstrap_requests_to_keys_query_response(bootstrap_requests);
+    bob.receive_keys_query_response(&TransactionId::new(), &kq_response).await.unwrap();
+
+    let custom_event_type = "m.hello";
+
+    let custom_content = json!({
+            "what": "foo",
+    });
+
+    let received = encrypt_to_device_helper(&alice, &bob, custom_event_type, custom_content).await;
+
+    assert_let!(ProcessedToDeviceEvent::Decrypted { encryption_info, .. } = received);
```

Comment:
Maybe I could assert a snapshot on the decrypted raw result?

Reply:
You can use Insta's support for redactions, I used it here:

https://github.com/matrix-org/matrix-rust-sdk/blob/789b82b012fadcde2489da17e25e2174ba8493e0/crates/matrix-sdk-crypto/src/olm/group_sessions/inbound.rs#L826-L831

This rewrites the random key with a static string.

More info: https://insta.rs/docs/redactions/

Reply:
Done, thx for the tip. https://github.com/matrix-org/matrix-rust-sdk/pull/4935/commits/25405c343d1e9f2e3ef492dd56c7b8bbf1ef449f